pandas/tests/arrays/string_/test_string.py

```diff
@@ -187,39 +292,57 @@ def test_comparison_methods_array(all_compare_operators):
     tm.assert_extension_array_equal(result, expected)
 
 
-def test_constructor_raises():
-    with pytest.raises(ValueError, match="sequence of strings"):
-        pd.arrays.StringArray(np.array(["a", "b"], dtype="S1"))
+def test_constructor_raises(cls):
+    if cls is pd.arrays.StringArray:
+        msg = "StringArray requires a sequence of strings or pandas.NA"
+    else:
+        msg = "Unsupported type '<class 'numpy.ndarray'>' for ArrowStringArray"
 
-    with pytest.raises(ValueError, match="sequence of strings"):
-        pd.arrays.StringArray(np.array([]))
+    with pytest.raises(ValueError, match=msg):
+        cls(np.array(["a", "b"], dtype="S1"))
 
-    with pytest.raises(ValueError, match="strings or pandas.NA"):
-        pd.arrays.StringArray(np.array(["a", np.nan], dtype=object))
+    with pytest.raises(ValueError, match=msg):
+        cls(np.array([]))
 
-    with pytest.raises(ValueError, match="strings or pandas.NA"):
-        pd.arrays.StringArray(np.array(["a", None], dtype=object))
+    with pytest.raises(ValueError, match=msg):
+        cls(np.array(["a", np.nan], dtype=object))
 
-    with pytest.raises(ValueError, match="strings or pandas.NA"):
-        pd.arrays.StringArray(np.array(["a", pd.NaT], dtype=object))
+    with pytest.raises(ValueError, match=msg):
+        cls(np.array(["a", None], dtype=object))
+
+    with pytest.raises(ValueError, match=msg):
+        cls(np.array(["a", pd.NaT], dtype=object))
 
 
 @pytest.mark.parametrize("copy", [True, False])
-def test_from_sequence_no_mutate(copy):
+def test_from_sequence_no_mutate(copy, cls, request):
+    if cls is ArrowStringArray:
+        reason = (
+            "ValueError: Unsupported type '<class 'numpy.ndarray'>' for "
+            "ArrowStringArray"
+        )
+        mark = pytest.mark.xfail(reason=reason)
+        request.node.add_marker(mark)
+
     nan_arr = np.array(["a", np.nan], dtype=object)
     na_arr = np.array(["a", pd.NA], dtype=object)
 
-    result = pd.arrays.StringArray._from_sequence(nan_arr, copy=copy)
-    expected = pd.arrays.StringArray(na_arr)
```

Comment:
I think here you can probably create this `expected` value differently for string dtype vs arrow string dtype (as it is only this aspect of the test that is failing, and not what is actually being tested)

Reply:
@jorisvandenbossche does the second part of the test make sense with copy=False

we changed ensure_string_array to not use pd.NA and copy=False by default. https://github.com/pandas-dev/pandas/pull/35259#discussion_r522013997